src/column-layout/with-css/index.tsx

```diff
@@ -0,0 +1,79 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+import clsx from 'clsx';
+import flattenChildren from 'react-keyed-flatten-children';
+import { useContainerQuery } from '../../internal/hooks/container-queries';
+import { InternalColumnLayoutProps } from '../interfaces';
+import styles from './styles.css.js';
+
+const isOdd = (value: number): boolean => value % 2 !== 0;
+
+export function calculcateCssColumnCount(
+  columns: number,
+  minColumnWidth: number,
+  containerWidth: number | null
+): number {
+  if (!containerWidth) {
+    return columns;
+  }
+
+  // First, calculate how many columns we can have based on the current container width and minColumnWidth.
+  const targetColumnCount = Math.min(columns, Math.floor(containerWidth / minColumnWidth));
+
+  // When we start wrapping into fewer columns than desired, we want to keep the number of columns even.
+  return Math.max(
+    1,
+    targetColumnCount < columns && isOdd(targetColumnCount) ? targetColumnCount - 1 : targetColumnCount
+  );
+}
+
+interface ColumnLayoutWithCSSProps
+  extends Pick<InternalColumnLayoutProps, 'minColumnWidth' | 'columns' | 'variant' | 'borders' | 'disableGutters'> {
+  children: React.ReactNode;
+}
+
+export default function ColumnLayoutWithCSS({
```

Comment:
After having another look, I find this name confusing. Because this instance uses container query hook, it is not pure CSS, so the name is confusing.

Possible alternatives for the naming: Legacy/Modern, Flex/Grid